addons/base_import/models/base_import.py

```diff
@@ -476,107 +563,225 @@ def _try_match_date_time(self, preview_values, options):
+                matched_fields.append(mapping_field_name)
+                match_field = {
+                    'field_path': [name for name in mapping_field_name.split('/')],
+                    'distance': -1  # Trick to force to keep that match during mapping deduplication.
+                }
             if not match_field:
-                match_field = [field['name'] for field in self._match_header(header, fields, options)]
-            matches[index] = match_field or None
-        return headers, matches
+                match_field = self._get_mapping_suggestion(header, fields_tree, header_types[(index, header)])
+            mapping_suggestions[(index, header)] = match_field or None
+
+        mapping_suggestions = self._deduplicate_mapping_suggestions(mapping_suggestions)
+        return mapping_suggestions
+
+    def _deduplicate_mapping_suggestions(self, mapping_suggestions):
+        """ This method is meant to avoid multiple columns to be matched on the same field.
+        Taking mapping_suggestions as input, it will check if multiple columns are mapped to the same field and will
+        only keep the mapping that has the smallest distance. The other columns that were matched to the same field are
+        removed from the mapping suggestions.
+
+        Hierarchy mapping is considered as advanced and is ignore during this deduplication process. We consider that
+        multiple mapping on hierarchy mapping will not occur often and due to the fact that :
+            when a non 'char/text' field is selected more than once in the UI, we keep only the last selected mapping,
+        this won't lead to any particular issues. The objective is to lighten the mapping suggestion process as much
+        as we can.
+
+        :param (dict) mapping_suggestions: { (column_index, header_name) : { 'field_path': [header_name], 'distance': levenshtein_distance }}
+        :return (dict) new_suggestions: reconstructed dict of mapping_suggestions where duplicate mapping have been removed.
+                    we drop the header_name in the dict keys as we won't need it anymore."""
+        new_suggestions = {}
+        suggestions_min_dist = {}
+        for header, suggestion in mapping_suggestions.items():
+            column_index = header[0]
+            if suggestion is None:
+                continue
+            if len(suggestion['field_path']) > 1:  # ignore hierarchy mapping
+                new_suggestions[column_index] = suggestion['field_path']
+                continue
+            field_name = suggestion['field_path'][0]
+            field_distance = suggestion['distance']
+            if field_distance < suggestions_min_dist.get(field_name, {}).get('distance', 1):
+                suggestions_min_dist[field_name] = {
+                    'distance': field_distance,
+                    'index': column_index
+                }
+
+        for field_name, values in suggestions_min_dist.items():
```

Comment:
Mmmh, how about:

```py
keys_to_remove = []
min_dist_per_field = {}
for header, suggestion in mapping_suggestions.items():
    if suggestion is None or len(suggestion['field_path']) > 1:
        continue

    field_name = suggestion['field_path'][0]
    field_distance = suggestion['distance']
    if field_name not in min_dist_per_field:
        min_dist_per_field[field_name] = {'distance': field_distance, 'header_key': header}
    elif min_dist_per_field[field_name] < field_distance:
        # this field is a duplicate and is further in distance that the one stored -> remove it
        keys_to_remove.append(header)
    elif field_distance < min_dist_per_field[field_name]:
        # this field is a duplicate but the one we stored previously is further in distance
        # -> remove the stored one and replace it with the new min
        keys_to_remove.append(header)
        min_dist_per_field[field_name] = {'distance': field_distance, 'header_key': header}

for key in keys_to_remove:
    del mapping_suggestions[key]

return mapping_suggestions
```

That way you find keys that are to remove and remove them, instead of building some kind of inverse dict and then rebuilding the original dict out of it.

Might not be better, maybe it's a preference thing, I personally don't like the double dict thing because it feels like extra overhead.

Reply:
vendu! I changed it a bit (second "elif" becomes a "else") and I also had to remove the distance from the mapping_suggestion. To keep the input param structure intact, I removed the distance to keep only the field_path at the same place I remove the header_name from mapping keys. (see parse_preview)